jobs/payment-jobs/tasks/electronic_funds_transfer_task.py

```diff
@@ -12,273 +12,150 @@
+        receipt_response = CFSService.apply_receipt(cfs_account, receipt_number, invoice_number)
+        receipt_json = receipt_response.json()
+        receipt = ReceiptModel(receipt_number=receipt_json.get('receipt_number', None),
+                               receipt_amount=receipt_balance_before_apply -
+                               float(receipt_json.get('unapplied_amount')),
+                               invoice_id=invoice.id,
+                               receipt_date=datetime.now(tz=timezone.utc))
+        receipt.flush()
+
+    @classmethod
+    def link_electronic_funds_transfers_cfs(cls):
+        """Replicate linked EFT's as receipts inside of CFS.
 
         Steps:
-        1. Find all pending EFT Short Names in a LINKED state.
+        1. Find all pending EFT Credit Invoice Links in PENDING status.
         2. Receipts created in CAS representing the EFT.
         3. Apply the receipts to the invoices.
         4. Notify mailer
         """
-        eft_short_names = cls._get_eft_short_names_by_status(EFTShortnameStatus.LINKED.value)
-        for eft_short_name in eft_short_names:
+        credit_invoice_links = cls._get_eft_credit_invoice_links_by_status(EFTCreditInvoiceStatus.PENDING.value)
+        for invoice, credit_invoice_link, cfs_account in credit_invoice_links:
             try:
-                current_app.logger.debug(f'Linking Electronic Funds Transfer: {eft_short_name.id}')
-                payment_account = PaymentAccountModel.find_by_auth_account_id(
-                    eft_short_name.auth_account_id)
-                cfs_account = CfsAccountModel.find_effective_or_latest_by_payment_method(payment_account.id,
-                                                                                         PaymentMethod.EFT.value)
-                eft_credit = EFTCreditModel.find_by_payment_account_id(payment_account.id)
-
-                payment = db.session.query(PaymentModel) \
-                    .join(PaymentAccountModel, PaymentAccountModel.id == PaymentModel.payment_account_id) \
-                    .join(EFTCreditModel, EFTCreditModel.payment_account_id == PaymentAccountModel.id) \
-                    .join(EFTShortNamesModel, EFTShortNamesModel.id == EFTCreditModel.short_name_id) \
-                    .filter(EFTShortNamesModel.id == eft_short_name.id).first()
-
-                receipt_number = payment.id
-
+                current_app.logger.debug(f'Create EFT Invoice Links in CFS - PayAccount: {invoice.payment_account_id}'
+                                         f'Id: {credit_invoice_link.id} - Amount: {credit_invoice_link.amount} '
+                                         f'- Invoice Id: {invoice.id}')
+                invoice_reference = InvoiceReferenceModel.find_by_invoice_id_and_status(
+                    credit_invoice_link.invoice_id, InvoiceReferenceStatus.ACTIVE.value
+                )
+                invoice_reference.status_code = InvoiceReferenceStatus.COMPLETED.value
```

Comment:
tried to set this early incase the invoice_reference isn't there.. it will fail right away